src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -182,6 +180,9 @@ internal string InferredUserName
     /// </summary>
     volatile Exception? _breakReason;
 
+    // Used by replication to change our cancellation behaviour on ColumnStreams.
+    internal bool LongRunningConnection { get; set; }
```

Comment:
We actually hadn't properly implemented this. Previously our ColumnStream Read/ReadAsync methods had no concept of command style `ensure` cancellation. So replication was (and to some extent still is) oscillating between the two behaviors, depending on whether it was waiting on a stream read or on an ensure. Since PgReader also leans on ColumnStream for things like byte reads, we need to support ensure behavior on the stream, but only in command context.